### PR TITLE
#1 - setup package skeleton

### LINKED ISSUE #1: Set up package skeleton
I would like to see all required steps from documentation (https://laravel.com/docs/11.x/packages) done in scope of this task. This includes:
- [x] autodiscoverable service provider
- [ ] default configuration file
- [x] readme introduction

Package will be published as `blumilksoftware/laravel-heatmap` in Packagist.

### PR DESCRIPTION
**Pull request by BarTracz**
Tested on Laravel application and it works. I have to add eloquent helper yet as mentioned in #2 but for now lets push it as it is. This should close #1. I need to write documentation, but I will do it after adding the eloquent helper, so I will open this as separate issue.

### CODE REVIEW

File: .github/workflows/check-pr-title.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  check-pr-title:
+    name: Check PR title
+    timeout-minutes: 10
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-22.04
+
+    steps:
+      - uses: blumilksoftware/action-pr-title@e05fc76a1cc45b33644f1de51218be43ac121dd0 # v1.2.0
```
Comment (EwelinaSkrzypacz) [inline .github/workflows/check-pr-title.yml]:
```suggestion
      - uses: blumilksoftware/action-pr-title@e05fc76a1cc45b33644f1de51218be43ac121dd0 # v1.2.0
      
```

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          coverage: none
+
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --no-interaction --no-suggest
+
+      - name: Run PHP linter
+        run: composer cs
+
+      - name: Execute tests
+        run: php vendor/bin/phpunit --configuration phpunit.xml
```
Comment (EwelinaSkrzypacz) [inline .github/workflows/php.yml]:
```suggestion
        run: php vendor/bin/phpunit --configuration phpunit.xml
        
```

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.idea
+.composer
+.phpunit.result.cache
+vendor
+composer.lock
+data.json
+heatmapOutput.json
+.php-cs-fixer.cache
```
Comment (EwelinaSkrzypacz) [inline .gitignore]:
```suggestion
.php-cs-fixer.cache

```

File: index.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        $startOfHour = $now->startOfHour();
+
+        $heatmapBuilder = new HeatmapBuilder(
+            now: $startOfHour,
+            periodInterval: $interval,
+            period: CarbonPeriod::create($startOfMonth, '1 day', $endOfMonth),
+            decorator: new TailwindDecorator("red"),
+            alignedToStartOfPeriod: true,
+            alignedToEndOfPeriod: true,
+            //timezone: new \Carbon\CarbonTimeZone('+2'),
```
Comment (EwelinaSkrzypacz) [inline index.php]: Do we need this comment here?

File: phpunit.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/|version|/phpunit.xsd"
+         bootstrap="vendor/autoload.php">
+    <testsuites>
+        <testsuite name="unit">
+            <directory>tests</directory>
+        </testsuite>
+    </testsuites>
+</phpunit>
```
Comment (EwelinaSkrzypacz) [inline phpunit.xml]:
```suggestion
</phpunit>

```

**Comment by krzysztofrewak** [comment]
Okay, I think there were some misunderstanding here. There's no reason to copypaste working package to new repository. What I expected was to create almost empty Laravel package that would require (in Composer) our base library. So if anyone would install `laravel-heatmap`, would get `heatmap` behind the curtain.

Please remove copypasted codebase and add `blumilksoftware/heatmap` to the Composer manifest.

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "blumilksoftware/laravel-heatmap",
+  "description": "Heatmap library for Laravel",
+  "type": "library",
+  "require": {
+    "php": "^8.2",
+    "nesbot/carbon": "^3.5",
+    "illuminate/support": "^11.11",
+    "blumilksoftware/heatmap": "dev-main"
```
Comment (JakubKermes) [inline composer.json]:
```suggestion
    "blumilksoftware/heatmap-builder": "*"
```

Comment (BarTracz) [reply composer.json]: i overlooked this, thanks

File: composer.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "blumilksoftware/laravel-heatmap",
+  "description": "Heatmap library for Laravel",
+  "type": "library",
+  "require": {
+    "php": "^8.2",
+    "nesbot/carbon": "^3.5",
+    "illuminate/support": "^11.11",
+    "blumilksoftware/heatmap-builder": "*"
```
Comment (krzysztofrewak) [inline composer.json]:
```suggestion
    "blumilksoftware/heatmap": "0.0.1"
```

File: composer.json [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        "Blumilk\\LaravelHeatmap\\LaravelHeatmapServiceProvider"
+      ]
+    }
+  },
+  "repositories": [
+    {
+      "type": "vcs",
+      "url": "https://github.com/blumilksoftware/heatmap.git"
+    }
+  ],
```
Comment (krzysztofrewak) [inline composer.json]: Now you can remove it.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: composer.json (hunk 7 above)
Comment (BarTracz) [reply composer.json]: Changed

File: composer.json (hunk 8 above)
Comment (BarTracz) [reply composer.json]: Removed

**Review by krzysztofrewak — APPROVED** [review]

**Review by JakubKermes — APPROVED** [review]

File: composer.json (hunk 6 above)
Comment (JakubKermes) [reply composer.json]: ok

**Pull request merged**